bundle/deployplan/action.go

```diff
@@ -38,30 +38,63 @@ const (
 )
 
 var ShortName = map[ActionType]string{
+	ActionTypeUpdateWithID: "update",
+}
+
+// FullName provides stable identifiers for actions including update_with_id.
+var FullName = map[ActionType]string{
 	ActionTypeNoop:         "noop",
 	ActionTypeResize:       "resize",
 	ActionTypeUpdate:       "update",
-	ActionTypeUpdateWithID: "update",
+	ActionTypeUpdateWithID: "update_with_id",
 	ActionTypeCreate:       "create",
 	ActionTypeRecreate:     "recreate",
 	ActionTypeDelete:       "delete",
 }
 
+var FullNameReverse = map[string]ActionType{}
+
+func init() {
+	for k, v := range FullName {
+		FullNameReverse[v] = k
+	}
+}
+
 func (a ActionType) IsNoop() bool {
 	return a == ActionTypeNoop
 }
 
 func (a ActionType) KeepsID() bool {
 	switch a {
-	case ActionTypeCreate, ActionTypeUpdateWithID, ActionTypeRecreate:
+	case ActionTypeCreate, ActionTypeUpdateWithID, ActionTypeRecreate, ActionTypeDelete:
 		return false
 	default:
 		return true
 	}
 }
 
-func (a ActionType) String() string {
-	return ShortName[a]
+// StringShort short version of action. Currently this only replaces "update_with_id" with "update".
```

Comment:
This actually doesn't seem to "replace" the name?

Reply:
I expanded the comment, please take a look - a94e08df8be038a483e3eceb4b655ebf6ddca5b3